### PR TITLE
Add ForcePull to BuildOverrides

### PR DESCRIPTION
**Pull request by coreydaley**
Adding ForcePull observation for BuildOverrides

https://issues.redhat.com/browse/BUILD-96

### CODE REVIEW

File: pkg/operator/configobservation/util.go [hunk 1]
```diff
@@ -21,6 +21,11 @@ func ObserveField(observedConfig map[string]interface{}, val interface{}, fieldN
 	switch v := val.(type) {
 	case int64, bool:
 		err = unstructured.SetNestedField(observedConfig, v, nestedFields...)
+	case *bool:
+		if skipIfEmpty && val == nil {
+			return nil
+		}
+		err = unstructured.SetNestedField(observedConfig, *v, nestedFields...)
```
Comment (adambkaplan) [inline pkg/operator/configobservation/util.go:28]: @coreydaley being a bit paranoid. If a caller provides `skipIfEmpty=false` and `val=nil`, we risk a nil pointer dereference panic, no?

I think our best option is to throw an error:

```
if !skipEmpty && val == nil {
    return fmt.Errorf("cannot observe nil value for bool pointer field %s", fieldName)
}
```

Comment (coreydaley) [reply pkg/operator/configobservation/util.go:31]: updated in current push

**Comment by adambkaplan** [comment]
/lgtm

Generated build pod has `forcePull: true` in the `BUILD` environment variable that is injected into all pods.
@coreydaley next step is to add an e2e test for this feature. We have a test suite in origin which alters the cluster configuration [1], test should set the build override and check that all build pods force pull.

[1] https://github.com/openshift/origin/blob/master/test/extended/builds/cluster_config.go#L344

**Pull request merged**